src/Badoo/SoftMocks.php

```diff
@@ -2298,6 +2316,31 @@ public function rewriteScalar_MagicConst_File()
         );
     }
 
+    public function rewriteExpr_New(\PhpParser\Node\Expr\New_ $newNode)
+    {
+        if ($newNode->class == 'ReflectionClass') {
+            return new \PhpParser\Node\Expr\New_(
```

Comment:
This approach is not desirable for several reasons:

1. You change the (pretty frequently used) class name that is created. Code that prints class names or examines the class name in any way will fail.
2. With this approach you will only be able to rewrite expressions like `new ReflectionClass();` and not `$class = 'ReflectionClass'; new $class();`. The latter might cause some subtle bugs as well.

Sadly my https://github.com/badoo/soft-mocks/pull/35 wasn't merged (yes, it is huge so I probably should've split it :)) but it should solve the same problem that you are trying to solve: it fixes reflection getFileName() as well as it allows xdebug to work and a bunch of other stuff by rewriting the file name in the file stream wrapper class.

Reply:
See discussion in https://github.com/badoo/soft-mocks/pull/6 for more details on why SoftMocks does not allow to mock `new` operator.

Reply:
Hi. Thank you for the answer. I was looking over forks and issues for sometime, but wasn't looking at merge requests at all:( I'll take a look at it.